x/dex/keeper/pool.go

```diff
@@ -133,9 +129,8 @@ func CalcGreatestMatchingRatio(
 
 // Mutates the Pool object and returns relevant change variables. Deposit is not commited until
 // pool.save() is called or the underlying ticks are saved; this method does not use any keeper methods.
-func (p *Pool) Deposit(maxAmount0 sdk.Int, maxAmount1 sdk.Int) (inAmount0 sdk.Int, inAmount1 sdk.Int, outShares sdk.Int) {
-	lowerReserve0 := &p.LowerTick0.TickData.Reserve0AndShares[p.FeeIndex].Reserve0
-	lowerTotalShares := &p.LowerTick0.TickData.Reserve0AndShares[p.FeeIndex].TotalShares
+func (p *Pool) Deposit(maxAmount0 sdk.Int, maxAmount1 sdk.Int, totalShares sdk.Int) (inAmount0 sdk.Int, inAmount1 sdk.Int, outShares sdk.Int) {
```

Comment:
Can we init the pool with the total number of shares (storing on the pool object), rather than passing it in here?

Reply:
Did this originally but it ends up resulting in a needless query to the bank keeper in the swap functions (don't need `totalShares` there).

Reply:
I'd say we shouldn't do share calculations within Deposit and Withdrawal then

Reply:
nvm carry on